benches/cub/device/histogram/even.cu

```diff
@@ -27,8 +39,18 @@ static void basic(nvbench::state &state,
   T lower_level = 0;
   T upper_level = elements;
 
-  if (Pattern == tbm::data_pattern::modulo_sequence)
-  {
+  // Whether bin computation in HistogramEven could potentially overflow
+  constexpr bool could_overflow = cuda::std::is_integral_v<T>;
+  
+  // Integral types are generated within [0, BinsCount) to avoid overflows
+  thrust::device_vector<T> input(elements);
+  if constexpr(could_overflow){
+    thrust::transform(data_gen.cbegin(), data_gen.cbegin() + elements, input.begin(), ModIntOp<T>{BinsCount});
```

Comment:
Can't we perform the operation in-place and avoid allocating another `O(n)` storage? 
> The input and output sequences may coincide, resulting in an in-place transformation.

Reply:
Good suggestion! Now transforming in-place 👍